### PR TITLE
Awsretry/cloudfront distribution

### PR DESCRIPTION
**Pull request by fnavarrodev**
##### SUMMARY
Adding AWSRetry.exponential_backoff when updating a cloudfront distribution.

Fixes https://github.com/ansible-collections/community.aws/issues/296

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
cloudfront_distribution

### CODE REVIEW

**Review by matejstrasek — APPROVED** [review]

File: plugins/modules/cloudfront_distribution.py [hunk 1]
```diff
@@ -1444,7 +1444,7 @@ def delete_distribution(client, module, distribution):
 
 def update_distribution(client, module, config, distribution_id, e_tag):
     try:
-        return client.update_distribution(DistributionConfig=config, Id=distribution_id, IfMatch=e_tag)['Distribution']
+        return client.update_distribution(aws_retry=True, DistributionConfig=config, Id=distribution_id, IfMatch=e_tag)['Distribution']
```
Comment (tremble) [inline plugins/modules/cloudfront_distribution.py:1447]: This is just one of the calls.  What I tend to find is that if one call is going to hit the issue, most of them will.  I'd recommend adding this to all of the `client.<something>` calls I think they're all between lines 1420 and 1471

**Review by tremble — DISMISSED** [review]

File: plugins/modules/cloudfront_distribution.py (hunk 1 above)
Comment (fnavarrodev) [reply plugins/modules/cloudfront_distribution.py:1447]: I don't have a way to test them all, but I will yes

File: plugins/modules/cloudfront_distribution.py [hunk 2]
```diff
@@ -2131,7 +2131,7 @@ def main():
         ]
     )
 
-    client = module.client('cloudfront')
+    client = module.client('cloudfront', retry_decorator=AWSRetry.exponential_backoff(retries=3, delay=3))
```
[the remark this answers is not in the document]
Comment (fnavarrodev) [reply plugins/modules/cloudfront_distribution.py:2134]: Ahhh ok I see.

**Comment by tremble** [comment]
One additional thing, please add a changelog entry: https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#changelogs

**Comment by markuman** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Review by alinabuzachis — APPROVED** [review]

**Review by markuman — APPROVED** [review]

**Pull request merged**